### PR TITLE
Added some rudimentary documentation about each of our services

### PR DESCRIPTION
**Pull request by wimax-grapl**
I'm adding some - very rudimentary, as you can see from all the TODOs - documentation about each of our services to our ReadTheDocs.

Would be great to get some improvements on the TODOs from people who actually know what they're talking about :) either as PR followups or just comments on this diff

### CODE REVIEW

File: docs/development/services.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+## Network Diagram
+![Network Diagram](network_diagram.png)
+
+
+## Grapl Services - main pipeline
+Unless otherwise specified, the input to a service is the output of the one described above it.
+
+
+### Graph Generator
+**Input:** Event logs (i.g. sysmon logs, osquery logs, Cloudtrail logs) from the customer's S3 bucket.
```
Comment (alexjmacdonald) [inline docs/development/services.md:11]: `i.g.` should be `e.g.`?

File: docs/development/services.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+**Work:** Turns these events into a standalone subgraph, independent of existing Dgraph state.
+
+**Output:** Stores the subgraph to S3 for Node Identifier.
+
+
+### Node Identifier
+**Work:** Identifies nodes in the incoming subgraph against DynamoDB.
```
Comment (alexjmacdonald) [inline docs/development/services.md:19]: I'm still coming to grips w/ the arch so pardon me if I'm off base -- DynamoDB holds the _canonical identities_ of the nodes?  is that the term we use to distinguish IDs in the incoming subgraph vs. the IDs in the master graph?

File: docs/development/services.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+### Graph Merger
+**Work:** Write the new edges and nodes from Node Identifier to Dgraph.
+
+**Output:** TODO (it looks like exactly the same as node identifier's output, just re-encoded?)
+
+
+### Analyzer Dispatcher
+**Work:** Determines which analyzers to execute against recently-merged graphs
```
Comment (alexjmacdonald) [inline docs/development/services.md:33]: no hyphen for adverbs

File: docs/development/services.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+
+### Analyzer Executor
+**Work:** Executes queries against Dgraph, and - upon a successful query hit - determines which lenses and risk nodes to create
+
+**Output:** A JSON dump of risks and lenses to be created by the Engagement Creator (see `analyzer_executor.py#emit_event` for specifics)
+
+
+### Engagement Creator
+**Work:** Simply writes the risks and lenses described in its input to Dgraph.
```
Comment (alexjmacdonald) [inline docs/development/services.md:45]: maybe s/write/append/ to describe the semantics (doesn't mutate existing structure, if that's correct)

File: docs/development/services.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+**Output:** Nothing is written to S3; the desired output is a mutation to Dgraph.
+
+
+## Other services
+### Metric Forwarder
+**Input:** 
+This services receives `stdout` from other lambdas via Cloudwatch Logs. We subscribe only to logs that contain the reserved keyword `MONITORING|`.
+
+**Work:** Metric Forwarder parses these log lines - which are basically `statsd` serialized metrics - and forwards them to Cloudwatch as metrics.
```
Comment (alexjmacdonald) [inline docs/development/services.md:55]: "basically statsd serialized metrics" = a different syntactic serialized representation of the same information?  idk what "basically" signifies here

File: docs/development/services.md (hunk 1 above)
Comment (wimax-grapl) [reply docs/development/services.md:11]: lol half ie half eg

File: docs/development/services.md (hunk 2 above)
Comment (wimax-grapl) [reply docs/development/services.md:19]: I'm not 100% sure myself, but that's my understanding - the dynamodb tables help deal with deduplication and correlation. 

(Ideally, these will be fleshed out or rewritten by experts.)

**Review by inickles-grapl — APPROVED** [review]

**Pull request merged**